pkg/envoy/ciliumenvoyconfig.go

```diff
@@ -63,9 +64,42 @@ type PortAllocator interface {
 	ReleaseProxyPort(name string) error
 }
 
-// ParseResources parses all supported Envoy resource types from CiliumEnvoyConfig CRD to Resources type
-// cecNamespace and cecName parameters, if not empty, will be prepended to the Envoy resource names.
-func ParseResources(cecNamespace string, cecName string, anySlice []cilium_v2.XDSResource, validate bool, portAllocator PortAllocator) (Resources, error) {
+// ListenersAddedOrDeleted returns 'true' if a listener is added or removed when updating from 'old'
+// to 'new'
+func (old *Resources) ListenersAddedOrDeleted(new *Resources) bool {
```

Comment:
Could probably use a unit test for this logic. Also, if we have a string-set data structure somewhere already, we could use it with listener names and do the set difference/intersection instead of manually calculating the difference with slices.

Reply:
Will add a unit test, but generally I'd like to keep predicates like this not allocation memory when not needed. Not that this is on a particularly hot path, but still...